Constify the Alica model

hendrik-skubch opened the pull request:
This requires a companion PR in supplementary:
https://github.com/carpe-noctem-cassel/supplementary/pull/7

- Cleans up the engine model and limits access to model elements to const versions.
- Various simplifications

hendrik-skubch commented on .clang-format:
```diff
@@ -0,0 +1,32 @@
+---
```
Yea, unfortunately there are two style files (ours was first :P).
afaik .clang-format is the default name by which clang-format finds the style file (I could be wrong).
We can change to the other, but we'd need to change the shell script and the commit hook.

StephanOpfer replied:
Well, it was first in the alica repository. We had ours in the cnc-msl repository... :P 
Ok, we adjust the commit hook and script.

hendrik-skubch commented on alica_engine/include/engine/Types.h:
```diff
@@ -0,0 +1,52 @@
+#pragma once
```
I wanted to abstract away form the implementation and indicate the semantics instead.
Yes, it clashes with RoleSet, found that out at some point.
Bag sounds pretty good to me, +1.

hendrik-skubch commented on .clang-format:
```diff
@@ -0,0 +1,32 @@
+---
+BasedOnStyle: Google
+Standard: Cpp11
+
```
Yes, this is tricky.

I'm fine with either, and my personal favourite is to use bracewrapping for classes, namespaces and methods, but inline braces for blocks (loops, if, ...), it strikes a balance between clarity and not thinning the code out too much.

However, the idea of bracewrapping isn't liked by some people here at all.
Can we each send someone to a thunderdome and jsut grab some popcorn?

hendrik-skubch commented:
- Renamed AgentSet and friends to AgentGrp (bag is confusing the ros guys).
- replaced .clang-format with a symlink
- Made some modifications to .clang-format as a compromise. Please check / comment / scream.
